src/script/components/MessagesList/VirtualizedMessagesList/useScrollMessages.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Wire
+ * Copyright (C) 2025 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {MutableRefObject, useEffect, useRef} from 'react';
+
+import {Virtualizer} from '@tanstack/react-virtual';
+
+import {Conversation} from '../../../entity/Conversation';
+import {StatusType} from '../../../message/StatusType';
+import {GroupedMessage, isMarker, Marker} from '../utils/messagesGroup';
+
+interface Props {
+  conversation: Conversation;
+  messages: (Marker | GroupedMessage)[];
+  highlightedMessage?: string;
+  userId: string;
+  conversationLastReadTimestamp: MutableRefObject<number>;
+}
+
+export const useScrollMessages = (
```

Comment:
please add comments to this file as well, what is generally this useScrollMessages used for & the 2 useEffect